### PR TITLE
Add TLS/SSL code samples and documentation. [API-1837]

### PR DESCRIPTION
**Pull request by mehmettokgoz**
- Two sample showing how to configure Go client with SSL enabled and Mutual Authentication enabled are added. 
- A new section to documentation is added to show the TLS/SSL options.

### CODE REVIEW

**Review by yuce — CHANGES_REQUESTED** [review]
Looks good, here are a few ideas...

File: doc.go [hunk 1]
```diff
@@ -226,6 +226,79 @@ In order to force using a specific date/time type, create a time.Time value and
...
+Hazelcast allows you to encrypt socket level communication between Hazelcast members and between Hazelcast clients and members, for end to end encryption.
+To use it, see the TLS/SSL for Hazelcast Members section: https://docs.hazelcast.com/hazelcast/latest/security/tls-ssl.html#tlsssl-for-hazelcast-members
+
+TLS/SSL for the Hazelcast Go client can be configured using the cluster.SSLConfig{} struct.
+This struct is accessible from the hazelcast.Config struct through Cluster.Network.SSL path.
+You can either edit the hazelcast.Config object or create a cluster.SSLConfig{} instance and then assign it to config.Cluster.Network.SSL.
+
+TLS/SSL for the Hazelcast Go client can be enabled/disabled using the SSLConfig.Enabled option.
+When this option is set to True, TLS/SSL will be configured with respect to the other SSL options.
+Setting this option to False will result in discarding the other SSL options.
```
[the remark this answers is not in the document]
Comment (mehmettokgoz) [reply doc.go]: Updated: https://github.com/hazelcast/hazelcast-go-client/pull/920/commits/352f2b627ac5f310f70d2be6f45c5e19456cab0e

File: doc.go [hunk 2]
```diff
@@ -226,6 +226,79 @@ In order to force using a specific date/time type, create a time.Time value and
...
+    The equivalent of a trust store is a cafile at the Go client side.
+
+You should set keyStore and trustStore before starting the members.
+
+Hazelcast allows you to encrypt socket level communication between Hazelcast members and between Hazelcast clients and members, for end to end encryption.
+To use it, see the TLS/SSL for Hazelcast Members section: https://docs.hazelcast.com/hazelcast/latest/security/tls-ssl.html#tlsssl-for-hazelcast-members
+
+TLS/SSL for the Hazelcast Go client can be configured using the cluster.SSLConfig{} struct.
+This struct is accessible from the hazelcast.Config struct through Cluster.Network.SSL path.
+You can either edit the hazelcast.Config object or create a cluster.SSLConfig{} instance and then assign it to config.Cluster.Network.SSL.
```
[the remark this answers is not in the document]
Comment (mehmettokgoz) [reply doc.go]: Removed at https://github.com/hazelcast/hazelcast-go-client/pull/920/commits/352f2b627ac5f310f70d2be6f45c5e19456cab0e.

File: doc.go [hunk 3]
```diff
@@ -226,6 +226,79 @@ In order to force using a specific date/time type, create a time.Time value and
...
+
+  - A Java keyStore is a file that includes a private key and a public certificate.
+    The equivalent of a key store is the combination of keyfile and certfile at the Go client side.
+  - A Java trustStore is a file that includes a list of certificates trusted by your application which is named certificate authority.
+    The equivalent of a trust store is a cafile at the Go client side.
+
+You should set keyStore and trustStore before starting the members.
+
+Hazelcast allows you to encrypt socket level communication between Hazelcast members and between Hazelcast clients and members, for end to end encryption.
+To use it, see the TLS/SSL for Hazelcast Members section: https://docs.hazelcast.com/hazelcast/latest/security/tls-ssl.html#tlsssl-for-hazelcast-members
```
[the remark this answers is not in the document]
Comment (mehmettokgoz) [reply doc.go]: Removed this section at https://github.com/hazelcast/hazelcast-go-client/pull/920/commits/352f2b627ac5f310f70d2be6f45c5e19456cab0e.

File: doc.go [hunk 4]
```diff
@@ -226,6 +226,79 @@ In order to force using a specific date/time type, create a time.Time value and
 	dateTimeValue := types.LocalDateTime(t)
 	dateTimeWithTimezoneValue := types.OffsetDateTime(t)
 
+# TLS/SSL
+
+One of the offers of Hazelcast is the TLS/SSL protocol which you can use to establish an encrypted communication across your cluster with key stores and trust stores.
```
[the remark this answers is not in the document]
Comment (mehmettokgoz) [reply doc.go]: Updated at https://github.com/hazelcast/hazelcast-go-client/pull/920/commits/352f2b627ac5f310f70d2be6f45c5e19456cab0e.

File: examples/ssl/simple/main.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	// Set the server name and select the protocol used in SSL communication, default is TLSv1_2
+	config.Cluster.Network.SSL.SetTLSConfig(&tls.Config{ServerName: "foo.bar", MinVersion: tls.VersionTLS13})
+	// Start a new Hazelcast client with SSL configuration.
+	client, err := hazelcast.StartNewClientWithConfig(ctx, config)
+	if err != nil {
+		panic(err)
+	}
+	defer client.Shutdown(ctx)
+	fmt.Println("Connection Successful!")
```
[the remark this answers is not in the document]
Comment (mehmettokgoz) [reply examples/ssl/simple/main.go]: Switched these two lines at https://github.com/hazelcast/hazelcast-go-client/pull/920/commits/352f2b627ac5f310f70d2be6f45c5e19456cab0e.

File: examples/ssl/simple/main.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/hazelcast/hazelcast-go-client"
+)
+
+func main() {
+	// Hazelcast server should be started with SSL enabled to use SSLConfig
+	ctx := context.TODO()
+	config := hazelcast.NewConfig()
```
[the remark this answers is not in the document]
Comment (mehmettokgoz) [reply examples/ssl/simple/main.go]: Updated at https://github.com/hazelcast/hazelcast-go-client/pull/920/commits/352f2b627ac5f310f70d2be6f45c5e19456cab0e.

**Review by yuce — APPROVED** [review]
Looks good!

**Pull request merged**